includes/controllers/class-message.php

```diff
@@ -67,6 +74,95 @@ public function __construct( $args = [] ) {
 		parent::__construct( $args );
 	}
 
+	/**
+	 * Fetches messages.
+	 *
+	 * @return WP_Rest_Response
+	 */
+	public function fetch_messages() {
+
+		// Check authentication.
+		if ( ! is_user_logged_in() ) {
+			return hp\rest_error( 401 );
+		}
+
+		// Get thread.
+		$thread_ids = hivepress()->request->get_context( 'message_thread_ids' );
+
+		// Check thread.
+		if ( ! $thread_ids ) {
+			return hp\rest_error( 400 );
+		}
+
+		// Get thread messages.
+		$thread_messages = Models\Message::query()->filter(
+			[
+				'id__in' => $thread_ids,
+			]
+		)->order( [ 'sent_date' => 'desc' ] )
+			->get();
+
+		// Set sender ID.
+		$sender_id = null;
+
+		foreach ( $thread_messages as $thread_message ) {
```

Comment:
Please pass the needed thread ID via JS, maybe with the data-render? It's possible if you change the request method to GET, then you can add?sender_id=123 at the end of the action URL, then looping over all the threads is not needed